docs/Inspection-of-yum-Repositories.md

```diff
@@ -0,0 +1,125 @@
+
+When a Red Hat system is inspected we need to handle yum repositories. There are two aspects we have to consider:
+* Reading the repository data from the system
+* Storing the repository data in the manifest file
+
+## Reading the Repository Data from the System
+
+This section presents different ways of accessing the yum repository data and lists their advantages and disadvantages.
+
+### Parse output of `yum -v repolist all`
+* Advantages:
+  * Lists all repos
+* Disadvantages:
+  * Parsing not trivial (line breaks depend on length of output)
+  * Not all repo attributes accessible
+
+### Parse the repo files in `/etc/yum.repos.d`
+* Advantages:
+  * list (almost) all repo attributes
+* Disadvantages:
+  * Repos from plugins missing (e.g. suse manager's repos)
+
+### Accessing the python api
+* Advantages:
+  * All repos and repo attributes accessible (you need to know the attribute's name, though)
+* Disadvantages:
+  * It's not clear how to get the repo's url using the api, yet
+  * Copying script to target needed
+
+### Calling `yum-config-manager`
+* Advantages:
+  * Lists all repo attributes (incl. defaults)
+* Disadvantages:
+  * Hard to parse
+  * Disabled repos missing
+
+## Storing the repository data in the system description
+
+There are different ways how the repositories can be stored in our data structure. This sections lists
+various approaches and their advantages and disadvantages.
+
+### Separate Sub-Sections for zypp and yum
+
+```
+* repositories
+  * zypp
+    * zypp 1
+    * zypp 2
+  * yum
+    * yum 1
+    * yum 2
+```
+
+* Advantages:
+  * Lists repos separately
+  * Different repos type can contain different attributes
+* Disadvantages:
+  * Schema migration needed
+
+### Flat Data Structure w/ Schema Change
+
+```
+* repositories
+  * zypp 1
+  * zypp 2
+  * yum 1
+  * yum 2
+```
+This approach includes storing the repository type.
+That means each entry in the repository list will get another attribute `type` which can be `zypp` or `yum`.
+
+* Advantages:
+  none
+* Disadvantages:
+  * Schema migration needed
+
+### Flat Data Structure w/o Schema Change
+
+```
+* repositories
+  * zypp 1
+  * zypp 2
+  * yum 1
+  * yum 2
+```
+This approach is similar to the approach above but doesn't include storing the repository type.
+All attributes of yum repos are mapped to zypp's attributes.
+
+* Advantages:
+  * No schema migration needed
+* Disadvantages:
+  * Not easy to build images using the available data
+
+## Considerations
+
+When reading the list of repositories we want to get all repositories, thus
+calling `yum-config-manager` or reading `/etc/yum.repos.d` doesn't work for us.
+`yum -v repolist all` is the prefered choice if we can deal with the line breaks in its output,
+because then we don't have to copy a file to the target system.
+
+The usecase for storing yum repositories is the inspection of a Red Hat system
+and creation of a human readable report. Building Red Hat images is out of scope
```

Comment:
Building Red Hat images is a less relevant use case. But there are more relevant use cases, such as migrating a Red Hat system to a SUSE system, or to compare a Red Hat system with a SUSE system. What are the implications of these?

When we only want to use the description as result of an inspection and not for anything else, we might want to think about if we need to flag the description somehow, so the tool can help the user to do the right things with it.